lib/vmpooler/pool_manager.rb

```diff
@@ -555,6 +577,10 @@ def check_pool(pool,
       end
     end
 
+    def vm_mutex(vmname)
+      @vm_mutex[vmname] || @vm_mutex[vmname] = Mutex.new
```

Comment:
will this grow forever or are the old mutex objects garbage collected at one point?

Reply:
From the research I've done about this ruby will clean up these objects as a part of the garbage collection process so long as they are no longer being used. Given the lifecycle of a VM I expect that these objects should not be around longer than the lifecycle of a given VM, which will allow ruby to clean them up.

Reply:
I talking specifically about the hash `vm_mutex`. Can we have a look throught the lifecycle of a few VMs and keep an eye on that hash to see if it only grows?

Reply:
Yes. That's where we will build up several hundred mutex objects. I did perform a test of creating a few hundred VMs, watching memory usage increased, destroy the VMs, and then watch memory usage return to the same value as startup time (over time, not right away, quickly reduced to 80%, after a few hours was back to startup time value). I think this is an indication that the object dereferencing and garbage collection are working as expected, but I'm happy to consider how we can test this further.